cmake/emu/nsim.cmake

```diff
@@ -1,44 +1,54 @@
 # SPDX-License-Identifier: Apache-2.0
-if(${CONFIG_SOC_NSIM_HS_SMP})
+if("${BOARD_DEBUG_RUNNER}" STREQUAL "mdb-nsim" OR "${BOARD_FLASH_RUNNER}" STREQUAL "mdb-nsim")
 # mdb is required to run nsim multicore targets
 find_program(
   MDB
   mdb
   )
+set(MDB_BASIC_OPTIONS -nooptions -nogoifmain -toggle=include_local_symbols=1)
 
-if(${CONFIG_SOC_NSIM_HS_SMP})
- set(MDB_ARGS mdb_hs_smp.args)
+# remove previous .sc.project folder which has temporary settings for MDB.
+set(MDB_OPTIONS ${CMAKE_COMMAND} -E rm -rf ${APPLICATION_BINARY_DIR}/.sc.project)
+if(CONFIG_MP_NUM_CPUS GREATER 1)
+  set(MULTIFILES ${MDB} -multifiles=)
+  foreach(val RANGE ${CONFIG_MP_NUM_CPUS})
+    if(val LESS CONFIG_MP_NUM_CPUS)
+      MATH(EXPR PSET_NUM "${CONFIG_MP_NUM_CPUS}-${val}")
+      MATH(EXPR CORE_NUM "${CONFIG_MP_NUM_CPUS}-${val}-1")
+      if (PSET_NUM GREATER 0)
+        list(APPEND MDB_OPTIONS &&)
+      endif()
+      list(APPEND MDB_OPTIONS ${MDB} -pset=${PSET_NUM} -psetname=core${CORE_NUM})
+      if (PSET_NUM GREATER 0)
+        list(APPEND MDB_OPTIONS -prop=download=2)
+        if (PSET_NUM GREATER 1)
+          set(MULTIFILES ${MULTIFILES}core${CORE_NUM},)
+        else()
+          set(MULTIFILES ${MULTIFILES}core${CORE_NUM})
+        endif()
+      endif()
+      list(APPEND MDB_OPTIONS ${MDB_BASIC_OPTIONS} -nsim @${BOARD_DIR}/support/${MDB_ARGS} ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME})
+    endif()
+  endforeach()
+  list(APPEND MDB_OPTIONS && NSIM_MULTICORE=1 ${MULTIFILES} -run -cl)
+else()
+ list(APPEND MDB_OPTIONS && ${MDB} ${MDB_BASIC_OPTIONS} -nsim @${BOARD_DIR}/support/${MDB_ARGS} -run -cl)
 endif()
 
 add_custom_target(run
   COMMAND
-  ${MDB} -pset=1 -psetname=core0 -prop=ident=0x00000050 -cmpd=soc
-  @${BOARD_DIR}/support/${MDB_ARGS} ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME} &&
-  ${MDB} -pset=2 -psetname=core1 -prop=download=2 -prop=ident=0x00000150 -cmpd=soc
-  @${BOARD_DIR}/support/${MDB_ARGS} ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME} &&
-  NSIM_MULTICORE=1 ${MDB} -multifiles=core0,core1 -cmpd=soc -run -cl
+  ${MDB_OPTIONS}
+  ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME}
   DEPENDS ${logical_target_for_zephyr_elf}
   WORKING_DIRECTORY ${APPLICATION_BINARY_DIR}
   USES_TERMINAL
-  )
+)
```

Comment:
It would be nice to add printing of MDB command here (i.e. if verbose mode is enabled).
As we construct the command in a quite complex way such logging will allow user to easily get this command to inspect it or run it manually.

Reply:
I add command to `COMMENT` which is the option of `add_custom_target`. When you run the command `make run` or `ninja run`, it will display the command.